From 3c89150dc3cc06afb905f07e215c34a540529261 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Fri, 24 May 2019 16:35:27 +0800
Subject: [PATCH] tty: serial: imx: replace residual readl/writel functions

Because of upstream commit 27c844261b87 ("serial: imx: add wrappers for writel and readl"),
all readl/writel should be replace with imx_uart_readl/writel.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/tty/serial/imx.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 0626983a64b9..eeed667af3fd 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1063,7 +1063,7 @@ static void imx_uart_break_ctl(struct uart_port *port, int break_state)
 
 static int imx_uart_setup_ufcr(struct imx_port *sport, unsigned int mode)
 {
-	unsigned int val;
+	unsigned int ufcr;
 	unsigned int rx_fifo_trig;
 
 	if (uart_console(&sport->port))
@@ -1072,9 +1072,10 @@ static int imx_uart_setup_ufcr(struct imx_port *sport, unsigned int mode)
 		rx_fifo_trig = RXTL_UART;
 
 	/* set receiver / transmitter trigger level */
-	val = readl(sport->port.membase + UFCR) & (UFCR_RFDIV | UFCR_DCEDTE);
-	val |= TXTL << UFCR_TXTL_SHF | rx_fifo_trig;
-	writel(val, sport->port.membase + UFCR);
+	ufcr = imx_uart_readl(sport, UFCR);
+	ufcr &= (UFCR_RFDIV | UFCR_DCEDTE);
+	ufcr |= TXTL << UFCR_TXTL_SHF | rx_fifo_trig;
+	imx_uart_writel(sport, ufcr, UFCR);
 	return 0;
 }
 
-- 
2.17.1

